src/utils/validateBurnedAmount/validateBurnedAmount.ts

```diff
@@ -0,0 +1,92 @@
+import type { Context } from '../../vms/context/model';
+import type { Transaction, UnsignedTx } from '../../vms/common';
+import type { EVMTx } from '../../serializable/evm';
+import { isImportExportTx as isEvmImportExportTx } from '../../serializable/evm';
+import { getBurnedAmountByTx } from '../getBurnedAmountByTx';
+import type { AvaxTx } from '../../serializable/avax';
+import { validateEvmBurnedAmount } from './validateEvmBurnedAmount';
+import type { GetUpgradesInfoResponse } from '../../info/model';
+import { isEtnaEnabled } from '../isEtnaEnabled';
+import { validateAvaxBurnedAmountEtna } from './validateAvaxBurnedAmountEtna';
+import { validateAvaxBurnedAmountPreEtna } from './validateAvaxBurnedAmountPreEtna';
+import {
+  isAvmBaseTx,
+  isExportTx as isAvmExportTx,
+  isImportTx as isAvmImportTx,
+} from '../../serializable/avm';
+import {
+  isAddDelegatorTx,
+  isAddValidatorTx,
+  isTransformSubnetTx,
+} from '../../serializable/pvm';
+
+const _getBurnedAmount = (tx: Transaction, context: Context) => {
+  const burnedAmounts = getBurnedAmountByTx(tx as AvaxTx | EVMTx);
+  return burnedAmounts.get(context.avaxAssetID) ?? 0n;
+};
+
+// Transactions that are deprecated or not implemented for Etna
+// Todo: remove isAvmBaseTx, isAvmExportTx and isAvmImportTx when avm dynmamic fee is implemented
+const isPreEtnaTx = (tx: Transaction) => {
```

Comment:
Can we rename this helper to something like `supportsStaticFee`? Also seems like x-chain dynamic fees are not in etna, so let's update this comment.

Reply:
<img width="597" alt="Screenshot 2024-10-10 at 12 06 01 PM" src="https://github.com/user-attachments/assets/cafbb73c-e91a-4c9e-9091-886392243fd3">

refactored to check supportedEtnaTx